File: dependencies/workmanager_apple/ios/Sources/workmanager_apple/LoggingDebugHandler.swift
import Foundation
import os

/**
 * A debug handler that outputs debug information to iOS's unified logging system.
 * Note: This class requires iOS 14.0 or later due to the use of os.Logger.
 */
@available(iOS 14.0, *)
public class LoggingDebugHandler: WorkmanagerDebug {
    private let logger = os.Logger(subsystem: "dev.fluttercommunity.workmanager", category: "debug")

    public override init() {}

    override func onTaskStatusUpdate(taskInfo: TaskDebugInfo, status: TaskStatus, result: TaskResult?) {
        switch status {
        case .scheduled:
            logger.debug("Task scheduled")
        case .started:
            logger.debug("Task started")
        case .completed:
            logger.debug("Task completed")
        case .failed:
            logger.error("Task failed")
        case .cancelled:
            logger.info("Task cancelled")
        case .retrying:
            logger.info("Task retrying")
        case .rescheduled:
            logger.info("Task rescheduled")
        }
    }

    override func onExceptionEncountered(taskInfo: TaskDebugInfo?, exception: Error) {
        logger.error("Task exception")
    }
}

File: dependencies/workmanager_android/android/src/main/kotlin/dev/fluttercommunity/workmanager/LoggingDebugHandler.kt
package dev.fluttercommunity.workmanager

import android.content.Context
import android.util.Log
import dev.fluttercommunity.workmanager.pigeon.TaskStatus

/**
 * A debug handler that outputs debug information to Android's Log system.
 */
class LoggingDebugHandler : WorkmanagerDebug() {
    companion object {
        private const val TAG = "WorkmanagerDebug"
    }

    override fun onTaskStatusUpdate(
        context: Context,
        taskInfo: TaskDebugInfo,
        status: TaskStatus,
        result: TaskResult?,
    ) {
        when (status) {
            TaskStatus.SCHEDULED -> Log.d(TAG, "Task scheduled")
            TaskStatus.STARTED -> Log.d(TAG, "Task started")
            TaskStatus.COMPLETED -> Log.d(TAG, "Task completed")
            TaskStatus.FAILED -> Log.e(TAG, "Task failed")
            TaskStatus.CANCELLED -> Log.w(TAG, "Task cancelled")
            TaskStatus.RETRYING -> Log.w(TAG, "Task retrying")
            TaskStatus.RESCHEDULED -> Log.w(TAG, "Task rescheduled")
        }
    }

    override fun onExceptionEncountered(
        context: Context,
        taskInfo: TaskDebugInfo?,
        exception: Throwable,
    ) {
        Log.e(TAG, "Task exception")
    }
}
